feat(ConversationHistory): added ability to edit convo history names (#624)
patternfly/chatbot: packages/module/patternfly-docs/content/extensions/chatbot/examples/UI/UI.md

```

> ### Drawer with editable conversations

Reviewer: ```suggestion
### Renaming conversations in history drawer
```

To make conversation names in the history drawer editable, pass the `isEditable` property to each conversation item that is intended or expected to be editable. When a conversation item is editable, you must ensure the following:

> - Each text input has a unique accessible name - this is handled automatically, but can be customized via the `inputAriaLabel` property on the conversation item.
> - The `onBlur` and `onKeyDown` callback handlers are passed in, typically to "exit" editing mode on blur and to handle the Enter or Escape keys on key down - in this example, `onBlur` and the Enter key via `onKeyDown` save the text input content, while Escape via `onKeyDown` cancels the edit and reverts to the original text.
> - The `onChange` callback handler is passed in to handle updating the text input value and any conversation state.
> - Focus is handled correctly when enabling and disabled editing mode - when `isEditing` becomes true, focus should be moved to the text input, and when the Enter or Escape keys are pressed and `isEditing` becomes false focus should be moved to the conversation actions dropdown toggle.

Reviewer: ```suggestion
- Each text input has a unique accessible name. This is handled automatically, but can be customized via the `inputAriaLabel` property on the conversation item.
- The `onBlur` and `onKeyDown` callback handlers are passed in, typically being used to exit editing mode on blur and to handle the **Enter** or **Escape** keys on key down. 
    - In this example, `onBlur` and the **Enter** key via `onKeyDown` save the text input content, while **Escape** via `onKeyDown` cancels the edit and reverts to the original text.
- The `onChange` callback handler is passed in to handle updating the text input value and any conversation state.
- Focus is handled correctly when enabling and disabled editing mode. 
    - When `isEditing` becomes `true`, focus should be moved to the text input. 
    - When the **Enter** or **Escape** keys are pressed and `isEditing` becomes `false`, focus should be moved to the conversation actions dropdown toggle.
```

```

### Drawer with editable conversations

> You can allow conversation items in the history drawer editable by implementing a modal that opens upon clicking a "Rename" (or similar) action. When doing so, you must ensure the following:

Reviewer: ```suggestion
You can allow users to edit conversation names in the history drawer by implementing a modal that opens upon clicking a "Rename" (or similar) action. When doing so, you must ensure the following:
```

> - When the modal opens, place focus at the end of the text input.

Reviewer: ```suggestion
- When the modal opens, focus is placed at the end of the text input.
```

- When the modal closes, focus goes back to the action toggle that was previously opened.

> - Changes can be canceled via the Escape key or clicking a "cancel" button.

Reviewer: ```suggestion
- Changes can be canceled via the **Escape** key or by clicking a "Cancel" button.
```

> - Changes can be saved via the Enter key or clicking a "save" button.

Reviewer: ```suggestion
- Changes can be saved via the **Enter** key or by clicking a "Save" button.
```
